### PR TITLE
make logger use onPush and async pipe for logs

### PR DESCRIPTION
**Pull request by krishi-vb**

### CODE REVIEW

File: src/app/services/toast.service.ts [hunk 1]
```diff
@@ -28,37 +28,47 @@ export class ToastService {
...
     this._show(
       { header, body, type: Event.SUCCESS },
       { ...this.DEFAULT_CONFIG_SUCCESS }
     );
-    this.toastLogs.push({ type: Event.SUCCESS, time: this.getTime() });
+    // this.toastLogs.push({ type: Event.SUCCESS, time: this.getTime() });
+
+    const log: ToastEventLog = { type: Event.SUCCESS, time: this.getTime() };
+    this._toastLogs$.next([...this._toastLogs$.getValue().concat([log])]);
   }
```
Comment (krishi-vb) [inline src/app/services/toast.service.ts]: wanted to understand how I can make `.next([...this._toastLogs$.getValue().concat([log])])` logic better (if my approach is right in the first place)

File: src/app/components/logger/logger.component.html [hunk 2]
```diff
@@ -2,13 +2,14 @@
...
-  <ng-template #renderLogs>
-    <div *ngFor="let event of toastEvents" class="logger-body">
-      A {{ event.type }} toast was served at {{ event.time }}
-    </div>
-  </ng-template>
+  <div *ngIf="toastLogs$ | async as logs" class="logger-body">
+    <div *ngIf="logs.length < 1; else renderLogs">No toasts served so far.</div>
+
+    <ng-template #renderLogs>
+      <div *ngFor="let log of logs">
```
Comment (anirudhvarma12) [inline src/app/components/logger/logger.component.html]: Ideally, should have a trackBy, even if its just index. 

[TrackBy](https://angular.io/api/core/TrackByFunction#description) function allows angular to optimise rendering of loops.

File: src/app/services/toast.service.ts (hunk 1 above)
Comment (anirudhvarma12) [reply src/app/services/toast.service.ts]: Better how? In terms of readability?

If yes, then you can just use a spread operator `[...this._toastLogs$.getValue(), log]`

**Review by anirudhvarma12 — CHANGES_REQUESTED** [review]

File: src/app/components/logger/logger.component.html (hunk 2 above)
Comment (krishi-vb) [reply src/app/components/logger/logger.component.html]: Thank you, made the change.

**Pull request merged**